src/verify.rs

```diff
@@ -95,21 +78,21 @@ fn prepare_hypothesis<'a>(state: &mut VerifyState, hyp: &'a Hyp) {
 
 /// Adds a named $e hypothesis to the prepared array.  These are not kept in the frame
 /// array due to infrequent use, so other measures are needed.
-fn prepare_named_hyp(state: &mut VerifyState, label: TokenPtr) -> Option<Diagnostic> {
+fn prepare_named_hyp(state: &mut VerifyState, label: TokenPtr) -> Result<()> {
```

Comment:
I kind of like this change, although I'll need to benchmark it — the optimizer is quite capricious about struct returns, and this _should_ be equivalent (all-zero for Ok, else 1usize followed by a Diagnostic) but I don't trust that.